packages/admin-ui/src/custom-collapsibe/use-collapsible.ts

```diff
@@ -0,0 +1,47 @@
+import { useCallback } from 'react'
```

Comment:
Just not sure about where the best place for this hook file is. Maybe we should have a `hooks` folder for these cases, where we have hooks that are not attached to a specific admin-ui component? So this could be at `src/hooks/use-collapsible.tsx` and its story file could be moved to `src/hooks/use-collapsible.stories.tsx`.

Reply:
AFAIK we're kind of using this pattern already. For instance, on the menu component we have a `menu.state.ts` file that just exports a hook (useMenuState): https://github.com/vtex/admin-ui/blob/dev/packages/admin-ui/src/menu/menu.state.ts. Same for the tab component (`tab.state.ts`): https://github.com/vtex/admin-ui/blob/dev/packages/admin-ui/src/tab/tab.state.ts

I would vote for renaming this file to `collapsible.state.ts` to keep consitent.

Reply:
Ah, got it! Let's follow the existing pattern then

Reply:
I agree. The file structure can be similar to this:
```
src
  |- collapsible
     |- collapsible.state.ts
     |- stories
     |- tests
     |- index.ts 
```